src/client/Microsoft.Identity.Client/Platforms/Features/WamBroker/AadPlugin.cs

```diff
@@ -360,38 +360,45 @@ public Task<WebTokenRequest> CreateWebTokenRequestAsync(WebAccountProvider provi
             return msalTokenResponse;
         }
 
-        public string MapTokenRequestError(WebTokenRequestStatus status, uint errorCode, bool isInteractive)
+        public Tuple<string, bool> MapTokenRequestError(WebTokenRequestStatus status, uint errorCode, bool isInteractive)
```

Comment:
I think you should return 3-item tuple instead: error_code, error_message, is_retryable. This way you can add some very detailed error messages from here for all the cases. In particular we need a very detailed error message for "no network" and for BT_SPURIOUS_ACTIVATION

Reply:
@bgavrilMS What details are needed for "no network"?

Reply:
Discussed offline. Added the error message.